src/modules/dashboard/BulkUpdateComponet.tsx

```diff
@@ -0,0 +1,163 @@
+import { MenuItem } from '@mui/material';
+
+import { UserContext } from 'App';
+import { useCategories, useDepartments } from 'modules/Category/category.hook';
+import { Button } from 'modules/shared/Button';
+import Select, { CustomSelect } from 'modules/shared/Select';
+import { useUsers } from 'modules/Ticket/ticket.hook';
+import { useCallback, useContext, useMemo, useState } from 'react';
+import { STATUS } from './constant';
+import { usePostBulkUpdate } from './dashboard.hooks';
+import { BulkUpload, PayloadBulkUpload, SelectedTicket } from './types';
+
+export const BulkUpdateComponent = ({
+  selectedTicketForBulkUpdate,
+  setSeletedTicketForBulkUpdate,
+  setFilterMenu,
+}: {
+  selectedTicketForBulkUpdate: SelectedTicket;
+  setSeletedTicketForBulkUpdate: (a: SelectedTicket) => void;
+  setFilterMenu: (a: any) => void;
+}) => {
+  const [payload, setpayloadload] = useState<BulkUpload>({
+    department_id: '',
+    category_id: '',
+    resolver_id: '',
+    status: '',
+  });
+  const statusOptions = useMemo(
+    () =>
+      selectedTicketForBulkUpdate.permited_transitions?.map((item) => ({
+        label: STATUS[item],
+        value: item,
+      })),
+    [selectedTicketForBulkUpdate]
+  );
+  const handleChange = useCallback(
+    (
+      name: 'department_id' | 'category_id' | 'resolver_id' | 'status',
+      value: string
+    ) => {
+      setpayloadload((p: BulkUpload) => ({ ...p, [name]: value }));
+    },
+    [setpayloadload]
+  );
+  const { userAuth } = useContext(UserContext);
+  const { data: categoriesList, isLoading: listFetching } = useCategories(
+    payload?.department_id
+  );
+  const categoryOptions = useMemo(() => {
+    return (
+      categoriesList?.map((cate) => ({
+        label: cate.name,
+        value: cate.id,
+      })) || []
+    );
+  }, [categoriesList]);
+
+  const [organizationId, setOrganizationId] = useState<number>(
+    userAuth?.organizations?.[0]?.id
+  );
+  const { data: departmentsList, isLoading: departmentsFetching } =
+    useDepartments(organizationId);
+
+  const deptOptions = useMemo(() => {
+    return (
+      departmentsList?.map((dept) => ({
+        label: dept.name,
+        value: dept.id,
+      })) || []
+    );
+  }, [departmentsList]);
+  const { data: usersList, isLoading: isFetchingUsers } = useUsers(
+    payload?.department_id
+  );
+  const userResolverList = useMemo(() => {
+    const list =
+      usersList?.map((emp) => {
```

Comment:
If possible do all manipulation of data & conversion in specific format inside the useQuery hook using select option

Reply:
this api also use to show resolver list table